mod/wikirate_source/spec/set/type/source/source_spec.rb

```diff
@@ -61,182 +85,184 @@
       expect(@source_page.format.render_titled).to eq(@source_page.format.render_titled_with_voting)
     end
 
-    it "renders open view with :custom_source_header to be true" do
+    it 'renders open view with :custom_source_header to be true' do
       expect(@source_page.format.render_open).to include(@source_page.format.render_header_with_voting)
     end
 
-    it "renders header view with :custom_source_header to be true" do
-      expect(@source_page.format.render_header  :custom_source_header=>true ).to include(@source_page.format.render_header_with_voting)
+    it 'renders header view with :custom_source_header to be true' do
+      expect(@source_page.format.render_header  custom_source_header: true ).to include(@source_page.format.render_header_with_voting)
     end
-    it "renders metric_import_link" do
+    it 'renders metric_import_link' do
       test_csv = File.open("#{Rails.root}/mod/wikirate_source/spec/set/type_plus_right/source/import_test.csv")
-      sourcepage = Card.create! :type_id=>Card::SourceID,:subcards=>{'+File'=>{ :file=>test_csv,:type_id=>Card::FileID}}
+      sourcepage = create_source file: test_csv
       html = sourcepage.format.render_metric_import_link
-      source_file = sourcepage.fetch :trait=>:file
-      expect(html).to have_tag("a",:with=>{:href=>"/#{source_file.cardname.url_key}?view=import"},:text=>"Import to metric values")
+      source_file = sourcepage.fetch trait: :file
+      expect(html).to have_tag("a",with: {href: "/#{source_file.cardname.url_key}?view=import"},text: 'Import to metric values')
     end
-    describe "original_icon_link" do
-      context "file source" do
-        it "renders upload icon" do
+    describe 'original_icon_link' do
+      context 'file source' do
+        it 'renders upload icon' do
           test_csv = File.open("#{Rails.root}/mod/wikirate_source/spec/set/type_plus_right/source/import_test.csv")
-          sourcepage = Card.create! :type_id=>Card::SourceID,:subcards=>{'+File'=>{ :file=>test_csv,:type_id=>Card::FileID}}
+          sourcepage = create_source file: test_csv
           html = sourcepage.format.render_original_icon_link
-          source_file = sourcepage.fetch :trait=>:file
-          expect(html).to have_tag("a",:with=>{:href=>source_file.file.url}) do
-            with_tag "i",:with=>{:class=>"fa fa-upload"}
+          source_file = sourcepage.fetch trait: :file
+          expect(html).to have_tag("a",with: {href: source_file.file.url}) do
+            with_tag "i",with: {class: 'fa fa-upload'}
           end
         end
       end
-      context "link source" do
-        it "renders globe icon" do
+      context 'link source' do
+        it 'renders globe icon' do
           html = @source_page.format.render_original_icon_link
-          expect(html).to have_tag("a",:with=>{:href=>@url}) do
-            with_tag "i",:with=>{:class=>"fa fa-globe"}
+          expect(html).to have_tag("a",with: {href: @url}) do
```

Comment:
Space missing after comma.<br>Space inside { missing.<br>Space inside } missing.<br>Prefer single-quoted strings when you don't need string interpolation or special symbols.